packages/teuchos/numerics/src/Teuchos_BLAS.cpp

```diff
@@ -113,11 +113,11 @@ namespace Teuchos {
 //resolving them when linking dlls.
 #ifdef _MSC_VER
 #  ifdef HAVE_TEUCHOS_COMPLEX
-     template BLAS<long int, std::complex<float> >;
-     template BLAS<long int, std::complex<double> >;
+     template class BLAS<long int, std::complex<float> >;
+     template class BLAS<long int, std::complex<double> >;
 #  endif
-     template BLAS<long int, float>;
-     template BLAS<long int, double>;
+     template class BLAS<long int, float>;
+     template class BLAS<long int, double>;
```

Comment:
This is inside of an ifdef for `_MSC_VER` so Trilinos PR testing will **not** test these changes.  But this looks like correct C++ code (and I am surprised that it compiled the way it was).